packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3151,7 +3151,19 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       _textInputConnection = _needsAutofill && currentAutofillScope != null
         ? currentAutofillScope!.attach(this, _effectiveAutofillClient.textInputConfiguration)
         : TextInput.attach(this, _effectiveAutofillClient.textInputConfiguration);
-      _updateSizeAndTransform();
+      // It has to be run in post-frame callback, because when the current
+      // function is executed, we may be in build phase, while the
+      // _updateSizeAndTransform may need layout phase information.
+      // See https://github.com/flutter/flutter/issues/126312 for more details.
+      if (canTouchLayoutPhaseData) {
+        _updateSizeAndTransform();
+      } else {
+        SchedulerBinding.instance.addPostFrameCallback((Duration _) {
+          if (_textInputConnection != null) {
+            _updateSizeAndTransform();
+          }
+        });
```

Comment:
It looks like this could potentially mess with the `_textInputConnection!.requestAutofill()` call below (around line 3181 in the `if (_needsAutofill)` check). I'm not sure if the case you're solving for here can happen at the same time as the one that was resolved by moving that call, but it'd be good to be careful we're not regressing an old issue here.

For reference:
* [PR 92945](https://github.com/flutter/flutter/pull/92945) introduced the requirement that `requestAutofill` came after setting the size and transform
  * A test was introduced, but it doesn't look like the test overlaps with your reproduction, so the combination of the two may be a new edge case
* [Issue 68571](https://github.com/flutter/flutter/issues/68571) is the issue that was resolved by moving the `requestAutofill` call and has reproduction steps that may be able to be combined with your own reproduction

If it turns out that the two cases can't overlap, could you add a note about why either here or as part of your comment above?

Reply:
Good catch! Then what about @justinmc 's suggestion here https://github.com/flutter/flutter/pull/126324#discussion_r1188934147? If we never opened it (instead of open it but not update size/transform), maybe we are safer?

Reply:
With what you've got right now (putting the entire call in a post frame callback), I'm not worried about the edge case since the `_updateSizeAndTransform` and `requestAutofill` will still be guaranteed to be in the required order, just at a different stage. Thanks!

Reply:
You are welcome!